L1Trigger/TrackerTFP/src/KalmanFilter.cc

```diff
@@ -0,0 +1,398 @@
+  void KalmanFilter::consume(const StreamsTrack& streamsTrack, const TTDTC::Streams& streamsStub) {
+    auto valid = [](const auto& frame){ return frame.first.isNonnull(); };
+    auto acc = [](int& sum, const auto& frame){ return sum += (frame.first.isNonnull() ? 1 : 0); };
+    int nTracks(0);
+    int nStubs(0);
+    const int offset = region_ * dataFormats_->numChannel(Process::kf);
+    for (int channel = 0; channel < dataFormats_->numChannel(Process::kf); channel++) {
+      const int channelTrack = offset + channel;
+      const StreamTrack& streamTracks = streamsTrack[channelTrack];
+      nTracks += accumulate(streamTracks.begin(), streamTracks.end(), 0, acc);
+      for (int layer = 0; layer < setup_->numLayers(); layer++) {
+        const int channelStub = channelTrack * setup_->numLayers() + layer;
+        const TTDTC::Stream& streamStubs = streamsStub[channelStub];
+        nStubs += accumulate(streamStubs.begin(), streamStubs.end(), 0, acc);
+      }
+    }
+    tracks_.reserve(nTracks);
+    stubs_.reserve(nStubs);
+    for (int channel = 0; channel < dataFormats_->numChannel(Process::kf); channel++) {
+      const int channelTrack = offset + channel;
+      const StreamTrack& streamTracks = streamsTrack[channelTrack];
+      vector<TrackKFin*>& tracks = input_[channel];
+      tracks.reserve(streamTracks.size());
+      for (int frame = 0; frame < (int)streamTracks.size(); frame++) {
+        const FrameTrack& frameTrack = streamTracks[frame];
+        if (frameTrack.first.isNull())
+          continue;
+        const auto end = find_if(next(streamTracks.begin(), frame + 1), streamTracks.end(), valid);
+        const int size = distance(next(streamTracks.begin(), frame), end);
+        tracks.insert(tracks.end(), size - 1, nullptr);
+        deque<StubKFin*> stubs;
+        for (int layer = 0; layer < setup_->numLayers(); layer++) {
+          const int channelStub = channelTrack * setup_->numLayers() + layer;
+          const TTDTC::Stream& streamStubs = streamsStub[channelStub];
+          for (int i = frame; i < frame + size; i++) {
+            const TTDTC::Frame& frameStub = streamStubs[i];
+            if (frameStub.first.isNull())
+              break;
+            stubs_.emplace_back(frameStub, dataFormats_, layer);
+            stubs.push_back(&stubs_.back());
+          }
+        }
+        tracks_.emplace_back(frameTrack, dataFormats_, vector<StubKFin*>(stubs.begin(), stubs.end()));
+        /*cout << "KFin " << tracks_.back().inv2R() << " " << tracks_.back().phiT() << endl;
+        cout << "KFin " << -.5 * frameTrack.first->rInv() << endl;*/
+        tracks.push_back(&tracks_.back());
+      }
+    }
+  }
+
+  // fill output products
+  void KalmanFilter::produce(TTDTC::Streams& acceptedStubs, StreamsTrack& acceptedTracks, TTDTC::Streams& lostStubs, StreamsTrack& lostTracks) {
+    auto put = [this](const deque<State*>& states, TTDTC::Streams& streamsStubs, StreamsTrack& streamsTracks, int channel) {
```

Comment:
Lambda functions are usually just a single line. If you don't want to upgrade this to a full function, then please add comments "Start of lambda functions" and "End of lambda functions" to start and end of it, to make it stand out a bit.